runtime/lua/vim/lsp/semantic_tokens.lua

```diff
@@ -585,6 +585,76 @@ function M.stop(bufnr, client_id)
   end
 end
 
+--- Return the semantic token(s) at the given position.
+--- If called without argument, returns the token under the cursor.
+---
+---@param bufnr number|nil Buffer number (0 for current buffer, default)
+---@param row number|nil Position row (default cursor position)
+---@param col number|nil Position column (default cursor position)
+---
+---@return table[]|nil tokens Table of tokens at position
+function M.get_tokens_at_pos(bufnr, row, col)
```

Comment:
What do you think of calling this `get` and move the args into an `opts` table, treating each option like an additive filter? Similar to `diagnostic.get`:  `get(bufnr, { lnum = ..., col = ... })` (not sure about row vs. lnum, there's afaik already both used)

The `tokens` is a bit redundant as it's already in the semantics_token module.
`show_tokens_at_cursor` could then be changed to either `show(tokens)` or `print(tokens)`. I'd tend to `print` to mirror `vim.pretty_print`. Show has some "special display" behavior in other places (.e.g `diagnostic.show` doesn't print diagnostics)

One would need to compose them to print: `M.print(M.get())`

Reply:
Hmm, not a fan, I have to admit, but will think about it.

Note that show uses `nvim_echo`  to style the tokens and not just prints them. That's the big selling point here.

Reply:
But I agree that we're in a bit of a bind here, API wise, between diagnostics and treesitter. Personally, I think that treesitter is the closer relative even though as an LSP module there's a relationship to diagnostics.

Reply:
Setting aside the `show` vs. `print` question, `.get()` seems like a vast improvement over `get_tokens_at_pos()` and I strongly prefer that. As @mfussenegger said having `tokens` in the function name is repetitive, as is `_at_pos()` since we can just use a table argument for the position.

Pre `vim.diagnostic` the diagnostic functions were something like `vim.lsp.diagnostic.get_line_diagnostics()` which has many of the same issues that we're discussing here. I think we should stick with the `get()` pattern which is versatile enough to cover many different use cases.

Reply:
But should the behavior be changed for that? Or still just return the one token (per server) for a single position?

Because I think there's a fundamental difference here between diagnostics (where what is more fundamental than where) and extmarks (where it's the opposite).

Reply:
To be clear, the difference to me here is that it might make sense to get all diagnostics, but it doesn't make sense to get all tokens/captures/extmarks.

Reply:
Absolutely no objection to shortening these to `get_at_pos` and `show_at_cursor`, though.